integration_tests/e2e/refer/programmeHistory.cy.ts

```diff
@@ -503,97 +488,73 @@ context('Programme history', () => {
-          courseParticipation = courseParticipationFactory.withCourseId().build()
-          path = referPaths.programmeHistory.details.show({
-            courseParticipationId: courseParticipation.id,
-            referralId: referral.id,
-          })
+        const path = referPaths.programmeHistory.details.show({
+          courseParticipationId: courseParticipationWithCourseId.id,
+          referralId: referral.id,
+        })
 
-          cy.task('stubParticipation', courseParticipation)
-          cy.task('stubPrisoner', prisoner)
-          cy.task('stubReferral', referral)
+        beforeEach(() => {
+          cy.task('stubParticipation', courseParticipationWithCourseId)
 
           cy.visit(path)
         })
 
         it('shows the details page with the form fields pre-populated', () => {
           const programmeHistoryDetailsPage = Page.verifyOnPage(ProgrammeHistoryDetailsPage, {
             course: courses[0],
-            courseParticipation,
+            courseParticipation: courseParticipationWithCourseId,
             person,
           })
-
           programmeHistoryDetailsPage.shouldHaveCorrectFormValues()
         })
       })
     })
   })
 
   describe('When removing from the programme history', () => {
-    const prisoner = prisonerFactory.build({
-      firstName: 'Del',
-      lastName: 'Hatton',
+    const path = referPaths.programmeHistory.delete({
+      courseParticipationId: courseParticipationWithCourseId.id,
+      referralId: referral.id,
     })
-    const person = personFactory.build({
-      currentPrison: prisoner.prisonName,
-      name: 'Del Hatton',
-      prisonNumber: prisoner.prisonerNumber,
-    })
-    const referral = referralFactory.started().build({ prisonNumber: person.prisonNumber })
-    const course = courseFactory.build()
-
-    let courseParticipation: CourseParticipation
-    let courseParticipationWithName: CourseParticipationWithName
-    let path: string
 
     beforeEach(() => {
-      courseParticipation = courseParticipationFactory
-        .withCourseId()
-        .build({ courseId: course.id, prisonNumber: person.prisonNumber })
-      courseParticipationWithName = { ...courseParticipation, name: course.name }
-      path = referPaths.programmeHistory.delete({
-        courseParticipationId: courseParticipation.id,
-        referralId: referral.id,
-      })
+      cy.task('stubParticipation', courseParticipationWithCourseId)
+      cy.task('stubCourse', courses[0])
 
-      cy.task('stubPrisoner', prisoner)
-      cy.task('stubReferral', referral)
-      cy.task('stubParticipation', courseParticipation)
-      cy.task('stubCourse', course)
-
-      cy.signIn()
       cy.visit(path)
     })
 
     it('shows the delete page', () => {
       const deleteProgrammeHistoryPage = Page.verifyOnPage(DeleteProgrammeHistoryPage, {
-        participationWithName: courseParticipationWithName,
+        participationWithName: courseParticipationWithCourseIdWithName,
```

Comment:
`courseIdAndName`? (sorry!)

Reply:
Ha, I thought that too but didn't do it for some reason